src/streamlink/plugin/plugin.py

```diff
@@ -323,5 +323,10 @@ def get_streams(self, *args, **kwargs):
     def _get_streams(self):
         raise NotImplementedError
 
+    def stream_title(self):
```

Comment:
The default should be handled in one place only, if it's the `Plugin` base class it might be better do something like this: 
```python
def stream_title(self):
    title = self._get_title()
    if title is not None:
        return title
    else:
        return self.url

def _get_title(self):
    return None
```
And the `DEFAULT_PLAYER_ARGUMENTS`/`create_format_args` stuff should be refactored. 

side note: to clarify; the `streams` method is the public method for getting the streams, it normalises the result of `_get_streams`. `Plugin` subclasses _must_ implement `_get_streams`.